pkg/scheduler/framework/types.go

```diff
@@ -195,10 +271,44 @@ func (s QueueingHint) String() string {
 // Resource represents the standard API resources such as Pod, Node, etc.
 // ActionType denotes the specific change such as Add, Update or Delete.
 type ClusterEvent struct {
-	Resource   GVK
+	Resource   Resource
 	ActionType ActionType
-	// Label describes this cluster event, only used in logging and metrics.
-	Label string
+
+	// label describes this cluster event.
+	// It's an optional field to control String(), which is used in logging and metrics.
+	// Normally, it's not necessary to set this field; only used for special events like UnschedulableTimeout.
+	label string
+}
+
+// Label is used for logging and metrics.
+func (ce ClusterEvent) Label() string {
+	if ce.label != "" {
+		return ce.label
+	}
+
+	return fmt.Sprintf("%v%v", string(ce.Resource), ce.ActionType)
+}
+
+// AllClusterEventLabels returns all possible cluster event labels given to the metrics.
+func AllClusterEventLabels() []string {
+	var labels []string
```

Comment:
```suggestion
	labels := []string{UnschedulableTimeout.Label()}
```
Then we don't need line 310.